python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -820,8 +825,17 @@ def _evaluate_sensor(
     if sensor_runtime_data.log_key:
         context.add_log_key(sensor_runtime_data.log_key)
 
+    if sensor_runtime_data.asset_events:
+        context.update_state(TickStatus.SKIPPED, has_emitted_asset_events=True)
     for asset_event in sensor_runtime_data.asset_events:
-        instance.report_runless_asset_event(asset_event)
+        # TODO: enable linking asset check evaluations, which do not have tags
+        if isinstance(asset_event, AssetCheckEvaluation):
+            instance.report_runless_asset_event(asset_event)
+        else:
+            event_with_tags = asset_event.with_tags(
+                merge_dicts(asset_event.tags or {}, {TICK_ID_TAG: context.tick_id})
+            )
+            instance.report_runless_asset_event(event_with_tags)
```

Comment:
is there anyway to guarantee exactly-once semantics here on the inserted events if the tick fails partway through, similar to what we currently do for runs and backfills that are emitted from sensors? https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_daemon/sensor.py#L1057-L1067

Imagining a situation where a daemon crashes partway through (or worse, crashes in a loop for whatever reason on the same tick) and the same event gets added over and over again

We could look for existing events with the tag, but i think the challenge there is that if the tick retries if will (I think) have a different tick ID.